### PR TITLE
fix(player): keep status changes from shifting the UI

### LINKED ISSUE #315: Player view: UI jumps around while streaming status is changing
### What happened?

Version `0.0.14` Obtanium

While using the player view the whole UI jumps around when the status changes (`buffering`, `playing from navidrome`, `from cache`).

Granted it's a minor issue :)

I assume the different states occupy different amounts of vertical space due to icon sizes.

https://github.com/user-attachments/assets/1c64eace-0452-433e-837c-4b99f7830d87

### Steps to reproduce

1. play a song
2. go into full player mode
3. use the scrubber to jump to different parts of a song to trigger the status changes

### Diagnostics

```text

```

### Install source

Obtainium

### What were you using?

- [ ] Local files
- [ ] Jellyfin
- [x] Navidrome / Subsonic
- [ ] Android Auto
- [ ] Chromecast / Cast
- [ ] Offline cache / downloads

### Logs (optional)

```text

```

### PR DESCRIPTION
**Pull request by TheZupZup**
## Summary

Fixes the full player layout jumping when playback moves between streaming, buffering, and cache states.

- reserves one fixed-height slot for the source, buffering, casting, and error indicators;
- centers every indicator inside that slot so their different icon and text metrics cannot move the progress bar or transport controls;
- keeps long playback errors on one line with an ellipsis instead of allowing the status area to grow;
- adds a widget regression test that transitions from `Playing from Navidrome` to `Buffering…` to `Playing from Cache` and asserts the slider stays at the same vertical position.

No playback, seeking, provider, cache, or streaming behavior changes.

Closes #315

## Validation

The focused widget regression test is included in this PR. GitHub Actions will run formatting, analysis, the full Flutter test suite, Android debug compilation, and release lint validation.

### CODE REVIEW

File: lib/features/player/player_screen.dart [hunk 1]
```diff
@@ -196,7 +196,13 @@ class _LiveControls extends ConsumerWidget {
     return Column(
       mainAxisSize: MainAxisSize.min,
       children: [
-        _SourceOrError(state: state),
+        // Every source/buffering/casting/error state shares one fixed-height
+        // slot. Their icon and text metrics differ slightly, but the progress
+        // bar and transport controls must never move when the state changes.
+        SizedBox(
+          height: 24,
+          child: Center(child: _SourceOrError(state: state)),
```
[the remark this answers is not in the document]
Comment (TheZupZup) [reply lib/features/player/player_screen.dart]: Good catch. I replaced the hard-coded 24 px slot with a stable height derived from the current `MediaQuery.textScalerOf(context)` and the largest status text style. It still has a 24 px minimum for the icons, but now grows when the user enables larger system text. I also added a widget test using a 2× text scale to verify the slot expands without overflow.

File: test/features/player/player_status_layout_test.dart [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        position: Duration(minutes: 2),
+      ),
+    );
+    await tester.pump();
+
+    expect(find.text('Buffering…'), findsOneWidget);
+    expect(
+      tester.getTopLeft(slider).dy,
+      moreOrLessEquals(directStreamY, epsilon: 0.01),
+    );
```
[the remark this answers is not in the document]
Comment (TheZupZup) [reply test/features/player/player_status_layout_test.dart:89]: You're right — the slider position was anchored by the bottom controls and could stay unchanged even before the fix. The regression test now measures the track title metadata above the live controls, which is the area that visibly shifted when the status row changed height. The test still covers the Navidrome → Buffering → Cache transition.
